splitio/sync/synchronizer.py

```diff
@@ -505,40 +503,71 @@ def __init__(self, split_synchronizers, split_tasks):
         """
         self._split_synchronizers = split_synchronizers
         self._split_tasks = split_tasks
+        self._backoff = Backoff(
+                                _ON_DEMAND_FETCH_BACKOFF_BASE,
+                                _ON_DEMAND_FETCH_BACKOFF_MAX_WAIT)
 
-    def sync_all(self, max_retry_attempts=-1):
+    def sync_all(self, till=None):
         """
         Synchronize all splits.
-        
-        :param max_retry_attempts: Not used, added for compatibility
         """
-        try:
-            self._split_synchronizers.split_sync.synchronize_splits(None)
-        except APIException as exc:
-            _LOGGER.error('Failed syncing splits')
-            raise APIException('Failed to sync splits') from exc
+        self._backoff.reset()
+        remaining_attempts = _ON_DEMAND_FETCH_BACKOFF_MAX_RETRIES
+        while remaining_attempts > 0:
+            remaining_attempts -= 1
+            try:
+                return self.synchronize_splits()
+            except APIException as exc:
+                _LOGGER.error('Failed syncing all')
+                _LOGGER.error(str(exc))
```

Comment:
is this needed?

Reply:
Yes, if there is any i/o or json error, it will show up with this line.